pkg/writer/junit_xml.go

```diff
@@ -0,0 +1,192 @@
+    You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+package writer
+
+import (
+	"encoding/xml"
+	"fmt"
+	"io"
+
+	"github.com/accurics/terrascan/pkg/policy"
+	"github.com/accurics/terrascan/pkg/results"
+	"github.com/accurics/terrascan/pkg/version"
+)
+
+const (
+	junitXMLFormat supportedFormat = "junit-xml"
+	testSuiteName  string          = "TERRASCAN_POLICY_SUITE"
+	testSuitesName string          = "TERRASCAN_POLICY_SUITES"
+)
+
+// JUnitTestSuites is a collection of JUnit test suites.
+type JUnitTestSuites struct {
+	XMLName  xml.Name `xml:"testsuites"`
+	Tests    int      `xml:"tests,attr"`
+	Name     string   `xml:"name,attr"`
+	Failures int      `xml:"failures,attr"`
+	Time     string   `xml:"time,attr"`
+	Suites   []JUnitTestSuite
+}
+
+// JUnitTestSuite is a single JUnit test suite which may contain many testcases.
+type JUnitTestSuite struct {
+	XMLName    xml.Name        `xml:"testsuite"`
+	Tests      int             `xml:"tests,attr"`
+	Failures   int             `xml:"failures,attr"`
+	Time       string          `xml:"time,attr"`
+	Name       string          `xml:"name,attr"`
+	Package    string          `xml:"package,attr"`
+	Properties []JUnitProperty `xml:"properties>property,omitempty"`
+	TestCases  []JUnitTestCase
+}
+
+// JUnitTestCase is a single test case with its result.
+type JUnitTestCase struct {
+	XMLName   xml.Name `xml:"testcase"`
+	Classname string   `xml:"classname,attr"`
+	Name      string   `xml:"name,attr"`
+	Severity  string   `xml:"severity,attr"`
+	Category  string   `xml:"category,attr"`
+	// omit empty time because today we do not have this data
+	Time        string            `xml:"time,attr,omitempty"`
+	SkipMessage *JUnitSkipMessage `xml:"skipped,omitempty"`
+	Failure     *JUnitFailure     `xml:"failure,omitempty"`
+}
+
+// JUnitSkipMessage contains the reason why a testcase was skipped.
+type JUnitSkipMessage struct {
+	Message string `xml:"message,attr"`
+}
+
+// JUnitProperty represents a key/value pair used to define properties.
+type JUnitProperty struct {
+	Name  string `xml:"name,attr"`
+	Value string `xml:"value,attr"`
+}
+
+// JUnitFailure contains data related to a failed test.
+type JUnitFailure struct {
+	Message  string `xml:"message,attr"`
+	Type     string `xml:"type,attr"`
+	Contents string `xml:",chardata"`
+}
+
+func newJunitTestSuites(summary results.ScanSummary) JUnitTestSuites {
+	return JUnitTestSuites{
+		Tests:    summary.TotalPolicies,
+		Name:     testSuitesName,
+		Failures: summary.ViolatedPolicies,
+		Time:     fmt.Sprint(summary.TotalTime),
```

Comment:
We use the Sprintf to format this better. Adding the unit with the quantity, "seconds" here, i think